common/src/main/java/org/cloudfoundry/autosleep/access/cloudfoundry/CloudFoundryApi.java

```diff
@@ -307,40 +330,45 @@ public String getHost(String routeId) throws CloudFoundryException {
                     .get(GetRouteRequest.builder()
                             .routeId(routeId)
                             .build())
-                    .get(Config.CF_API_TIMEOUT);
+                    .block(Config.CF_API_TIMEOUT);
             RouteEntity routeEntity = response.getEntity();
             String route = routeEntity.getHost() + routeEntity.getPath();
             log.debug("route =  {}", route);
-
-            GetDomainResponse domainResponse = cfClient.domains()
-                    .get(GetDomainRequest.builder()
-                            .domainId(routeEntity.getDomainId())
-                            .build())
-                    .get(Config.CF_API_TIMEOUT);
-            log.debug("domain = {}", domainResponse.getEntity());
-            return route + "." + domainResponse.getEntity().getName();
+            String domainId = routeEntity.getDomainId();
+            String domain = getDeprecatedDomain(domainId);
+            return route + "." + domain;
         } catch (RuntimeException r) {
             throw new CloudFoundryException(r);
         }
     }
 
+    private String getDeprecatedDomain(String domainId) {
+        GetDomainResponse domainResponse = cfClient.domains()
```

Comment:
Question: We are still using the deprecated API. Do you think we could use the shared/private domains api to figure out the domain?